website_and_docs/content/documentation/webdriver/troubleshooting/logging.en.md

```diff
@@ -9,70 +9,165 @@ description: >
+<p>For example:</p>
+<pre>
 > 2022-12-24 16:07:09 WARN Selenium [DEPRECATION] [:jwp_caps] \`Capabilities#version=\` is deprecated. Use \`Capabilities#browser_version=\` instead.
-
-Because these items can get annoying, we've provided an easy way to turn them off.
-
-To turn off a specific warning, set the ID to ignore:
-```ruby
-Selenium::WebDriver.logger.ignore(:jwp_caps)
-```
-It accepts an `Array` to turn off multiple IDs:
-```ruby
-Selenium::WebDriver.logger.ignore(%i[jwp_caps pause pauses])
-```
-To turn off all *deprecation* notices:
-```ruby
-Selenium::WebDriver.logger.ignore(:deprecations)
-```
-
-**INFO**
-
-This is where the most useful information gets logged. Selenium logs the endpoints and payloads
+</pre>
+<br>
+<p>Because these items can get annoying, we've provided an easy way to turn them off.</p>
+
+<p>To turn off a specific warning, set the ID to ignore:</p>
+<pre>
+  Selenium::WebDriver.logger.ignore(:jwp_caps)
+</pre>
+<br>
+<p>It accepts an `Array` to turn off multiple IDs:</p>
+<pre>
+  Selenium::WebDriver.logger.ignore(%i[jwp_caps pause pauses])
+</pre>
+<br>
+<p>To turn off all <i>deprecation</i> notices:</p>
+<pre>
+  Selenium::WebDriver.logger.ignore(:deprecations)
+</pre>
+<br>
+<p><b>INFO</b></p>
+
+<p>This is where the most useful information gets logged. Selenium logs the endpoints and payloads
 sent to and received from the driver or server. This is a great way to see what Selenium is actually
 doing under the hood, and can be used to determine if it is Selenium code or driver code that
 is causing a problem. (Unfortunately, we can't blame the driver if Selenium is sending incorrect syntax).
+</p>
+
+<p><b>DEBUG</b></p>
+
+<p>This is less useful information where we log things about the servers and the sockets, and header information, etc.
+Debug mode is set if either <i>`$DEBUG`</i> is true or <i>`ENV['DEBUG']`</i> has a value.</p>
+{{< /tab >}}
+
+{{< tab header="JavaScript" >}}
+<p>JS has implemented logging APIs. It has also added LogManager to maintain a collection of loggers.</p>
+<p><b>1. Getting a logger can be done in two ways:</b></p>
+<pre>
+  const logging = require('selenium-webdriver/lib/logging')
+</pre>
+<br>
+<p>First method using manager</p>
+<pre>
+  mgr = new logging.LogManager()
+  logger = mgr.getLogger('')
+</pre>
+<br>
+<p>Second method without manager</p>
+<pre>
+  logger = logging.getLogger('')
+</pre>
+<br>
+<p>Loggers class uses hierarchical, dot-delimited namespaces (e.g. "" > "webdriver" > "webdriver.logging"). For example, the following code will give a hierarchy of loggers:</p>
+<pre>
+  logger = mgr.getLogger('foo.bar.baz')
+</pre>
+<br>
+<p>"" (parent of) > "foo" (parent of) > "foo.bar" (parent of) > "foo.bar.baz" (=logger)</p>
+
+<p><b>2. Logger Level:</b></p>
+
+<p>The basic levels for JS loggers are: <i>`OFF`, `SEVERE`, `WARNING`, `INFO`, `DEBUG`, `FINE`, `FINER`, `FINEST`, `ALL`</i>.<br>
+To log messages at the given level pass the level in the <i>`log()`</i> method:</p>
```

Comment:
How to log messages is important for writing Selenium code, but this page is intended to be for users who need to look at the logs coming from Selenium.

Reply:
Thank you for the clarity. This helped much. I was focussing on users' logging too. But now I have changed the approach to only target logs which Selenium generates.